src/Mod/Path/PathScripts/PathSurfaceSupport.py

```diff
@@ -933,6 +886,53 @@ def _calculateOffsetValue(self, isHole, isVoid=False):
 
         return offset
 
+    def findUnifiedRegions(
+            self,
+            shapeAndIndexTuples,
+            useAreaImplementation=True):
+        """Wrapper around area and wire based region unification
+        implementations."""
+        PathLog.debug('findUnifiedRegions()')
+        # Allow merging of faces within the LinearDeflection tolerance.
+        tolerance = self.obj.LinearDeflection.Value
+        # Default: normal to Z=1 (XY plane), at Z=0
+        try:
+            # Use Area based implementation
+            shapes = Part.makeCompound([t[0] for t in shapeAndIndexTuples])
+            outlineShape = PathUtils.getOffsetArea(
+                shapes,
+                # Make the outline very slightly smaller, to avoid creating
+                # small edges in the cut with the hole-preserving projection.
+                0.0 - tolerance / 10,
+                removeHoles=True,  # Outline has holes filled in
+                tolerance=tolerance,
+                plane=self.wpc)
+            projectionShape = PathUtils.getOffsetArea(
+                shapes,
+                # Make the projection very slightly larger
+                tolerance / 10,
+                removeHoles=False,  # Projection has holes preserved
+                tolerance=tolerance,
+                plane=self.wpc)
+            internalShape = outlineShape.cut(projectionShape)
+            # Filter out tiny faces, usually artifacts around the perimeter of
+            # the cut.
+            minArea = (10 * tolerance)**2
+            internalFaces = [
+                f for f in internalShape.Faces if f.Area > minArea
+            ]
+            if internalFaces:
+                internalFaces = Part.makeCompound(internalFaces)
+            return ([outlineShape], [internalFaces])
+        except Exception as e:
+            PathLog.warning(
+                f" getOffsetArea failed: {e}; Using FindUnifiedRegions.")
```

Comment:
`f"strings"` are not valid in Python 2, so this makes the unit tests fail. Actually, `f"strings"` are not supported by all versions of Python 3, only from 3.6 onwards, I think. So, it's better to not use them at the moment.

Reply:
Oh, didn't realize that Python 2 was still supported. Will change.

Reply:
Essentially once v0.19 is released we can drop support for Python 2 completely. However, I would still advice against using extremely new functionality of Python 3.

Reply:
Python 3.6 was released in December of 2016.  I'm not sure I'd qualify that as 'extremely new' but still better safe than sorry.

Reply:
Right. I mentioned this the last time somebody added fstrings to Path. Apparently, Ubuntu 16.04 still has Python 3.5 so even in Linux, some systems may not have the latest Python. And by extremely new features, I don't mean Python 3.6, obviously, I mean something that is just added in 3.9, or 3.10, something like that.